extensions/blocks/calendly/edit.js

```diff
@@ -0,0 +1,367 @@
+/**
+ * External Dependencies
+ */
+import classnames from 'classnames';
+import { isEqual } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	BlockControls,
+	BlockIcon,
+	InspectorControls,
+	PanelColorSettings,
+} from '@wordpress/block-editor';
+import {
+	Button,
+	ExternalLink,
+	Notice,
+	PanelBody,
+	Placeholder,
+	ToggleControl,
+	Toolbar,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { __, _x } from '@wordpress/i18n';
+import { ENTER, SPACE } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+import icon from './icon';
+import attributeDetails, { getValidatedAttributes } from './attributes';
+import SubmitButton from '../../shared/submit-button';
+
+const getUrlAndStyleFromEmbedCode = embedCode => {
+	if ( embedCode.indexOf( 'http' ) === 0 ) {
+		return {
+			style: 'inline',
+			url: embedCode,
+		};
+	}
+
+	let urlFromRegex = embedCode.match( / *data-url *= *["']?([^"']*)/i );
+	if ( urlFromRegex && urlFromRegex[ 1 ] && urlFromRegex[ 1 ].indexOf( 'http' ) === 0 ) {
+		return {
+			style: 'inline',
+			url: urlFromRegex[ 1 ],
+		};
+	}
+
+	urlFromRegex = embedCode.match( / *Calendly\.initPopupWidget\({ *url: *["']?([^"']*)/i );
+	if ( urlFromRegex && urlFromRegex[ 1 ] && urlFromRegex[ 1 ].indexOf( 'http' ) === 0 ) {
+		return {
+			style: 'link',
+			url: urlFromRegex[ 1 ],
+		};
+	}
+
+	urlFromRegex = embedCode.match( / *Calendly\.initBadgeWidget\({ *url: *["']?([^"']*)/i );
+	if ( urlFromRegex && urlFromRegex[ 1 ] && urlFromRegex[ 1 ].indexOf( 'http' ) === 0 ) {
+		return {
+			style: 'link',
+			url: urlFromRegex[ 1 ],
+		};
+	}
+};
+
+const getNewAttributesFromUrl = ( { url, style } ) => {
+	const attributes = { style };
+	const urlObject = new URL( url );
+	attributes.url = urlObject.origin + urlObject.pathname;
+
+	if ( ! urlObject.search ) {
+		return attributes;
+	}
+
+	const searchParams = new URLSearchParams( urlObject.search );
```

Comment:
Same IE11 polyfill issue as https://github.com/Automattic/jetpack/pull/14220/files#diff-8f69b389af6ef75b5a12cd4653cd595cR84 :-) Another option instead of the polyfill would be https://www.npmjs.com/package/url which calypso uses in some places for parsing urls.